### PR TITLE
fixed resetting start time bug

### PR DESCRIPTION
**Pull request by mcariatm**
### What would you like to be added:
Update status using client.Status().Update(..)

### Why is this needed:
we reset StartTime to allow updating the nodeSelectors on the job. However StartTime is in status, which we can update via a status update call only

### Issue Number:
#157

/kind bug

### CODE REVIEW

File: pkg/controllers/jobset_controller.go [hunk 1]
```diff
@@ -255,10 +255,13 @@ func (r *JobSetReconciler) resumeJobSetIfNecessary(ctx context.Context, js *jobs
 				log.Error(nil, "job missing ReplicatedJobName label")
 			}
 			job.Spec.Suspend = pointer.Bool(false)
-			job.Status.StartTime = nil
 			if err := r.Update(ctx, job); err != nil {
 				return err
 			}
+			job.Status.StartTime = nil
```
Comment (ahg-g) [inline pkg/controllers/jobset_controller.go:261]: please check if StartTime is already nil and avoid the update if it is

Comment (mcariatm) [reply pkg/controllers/jobset_controller.go:261]: In reconcile we call this function only if the job is suspended. But ok.

File: pkg/controllers/jobset_controller.go [hunk 2]
```diff
@@ -255,10 +255,16 @@ func (r *JobSetReconciler) resumeJobSetIfNecessary(ctx context.Context, js *jobs
...
 			if err := r.Update(ctx, job); err != nil {
 				return err
 			}
+			if job.Status.StartTime == nil {
+				continue
+			}
+			job.Status.StartTime = nil
+			if err := r.Status().Update(ctx, job); err != nil {
+				return err
+			}
```
Comment (ahg-g) [inline pkg/controllers/jobset_controller.go:267]: this is less intuitive than simply encapsulate the logic under the condition:
```suggestion
			if job.Status.StartTime != nil {
			  job.Status.StartTime = nil
			  if err := r.Status().Update(ctx, job); err != nil {
			 	return err
			  }
			}
```

File: pkg/controllers/jobset_controller.go [hunk 3]
```diff
@@ -255,10 +255,15 @@ func (r *JobSetReconciler) resumeJobSetIfNecessary(ctx context.Context, js *jobs
...
-			job.Status.StartTime = nil
 			if err := r.Update(ctx, job); err != nil {
 				return err
 			}
+			if job.Status.StartTime != nil {
+			  job.Status.StartTime = nil
+			  if err := r.Status().Update(ctx, job); err != nil {
+			 	return err
+			  }
+			}
```
Comment (ahg-g) [inline pkg/controllers/jobset_controller.go:266]: @mcariatm it seems that there is a missing bracket

Comment (mcariatm) [reply pkg/controllers/jobset_controller.go:266]: no, it was due to the indentation being too short.

Comment (mcariatm) [reply pkg/controllers/jobset_controller.go:266]: should be ok now....

**Pull request merged**